core/src/test/java/org/infinispan/xsite/irac/IracXsiteStatsTest.java

```diff
@@ -0,0 +1,470 @@
+      return CLUSTER_SIZE;
+   }
+
+   public void testQueueSizeStats(Method method) throws Exception {
+
+      MBeanServer mBeanServer = mBeanServerLookup.getMBeanServer();
+      ObjectName iracManager1 = getCacheObjectName(DOMAIN_NAME + 0, getDefaultCacheName() + "(dist_sync)",
+            "XSiteStatistics");
+      assertTrue(mBeanServer.isRegistered(iracManager1));
+
+      int queueSize = (Integer) mBeanServer.getAttribute(iracManager1, "queueSize");
+      assertEquals(queueSize, (int) 0);
+
+      // The initial state transfer uses cache commands, so it also increases the
+      ManualRpcManager rpcManager = rpcManagerList.get(0);
+      BlockedRequest req = rpcManager.block();
+
+      cache(0, 0).put("key", "value");
+
+      req.awaitRequest();
+      assertEquals((int) mBeanServer.getAttribute(iracManager1, "queueSize"), 1);
+
+      rpcManager.unblock();
+      eventuallyAssertInAllSitesAndCaches(cache -> Objects.equals("value", cache.get("key")));
+
+      assertEquals((int) mBeanServer.getAttribute(iracManager1, "queueSize"), 0);
+
+   }
+
+   public void testNoOfConflictsStats() throws Exception {
+
+      MBeanServer mBeanServer = mBeanServerLookup.getMBeanServer();
+      ObjectName iracManager1 = getCacheObjectName(DOMAIN_NAME + 0, getDefaultCacheName() + "(dist_sync)",
+            "XSiteStatistics");
+      ObjectName iracManager2 = getCacheObjectName(DOMAIN_NAME + 10, getDefaultCacheName() + "(dist_sync)",
+            "XSiteStatistics");
+      assertTrue(mBeanServer.isRegistered(iracManager1));
+      assertTrue(mBeanServer.isRegistered(iracManager2));
+
+      // The initial state transfer uses cache commands, so it also increases the
+      long noOfConflicts = (Long) mBeanServer.getAttribute(iracManager1, "NoOfConflicts");
+
+      assertEquals(noOfConflicts, (long) 0);
+
+      createConflict(Boolean.FALSE);
+
+      noOfConflicts = (Long) mBeanServer.getAttribute(iracManager2, "NoOfConflicts");
+      assertEquals(noOfConflicts, (long) 1);
+
+      // now reset statistics
+      mBeanServer.invoke(iracManager1, "resetStatistics", new Object[0], new String[0]);
+      assertEquals(mBeanServer.getAttribute(iracManager1, "noOfConflicts"), (long) 0);
+
+      mBeanServer.invoke(iracManager2, "resetStatistics", new Object[0], new String[0]);
+      assertEquals(mBeanServer.getAttribute(iracManager2, "noOfConflicts"), (long) 0);
+
+      mBeanServer.setAttribute(iracManager1, new Attribute("StatisticsEnabled", Boolean.FALSE));
+      mBeanServer.setAttribute(iracManager2, new Attribute("StatisticsEnabled", Boolean.FALSE));
+      assertEquals(mBeanServer.getAttribute(iracManager1, "noOfConflicts"), (long) -1);
+
+      // reset stats enabled parameter
+      mBeanServer.setAttribute(iracManager1, new Attribute("StatisticsEnabled", Boolean.TRUE));
+      mBeanServer.setAttribute(iracManager2, new Attribute("StatisticsEnabled", Boolean.TRUE));
+
+   }
+
+   public void testNumberOfConflictLocalWins() throws Exception {
```

Comment:
you can optimize the tests and remove `testNumberOfConflictLocalWins` and `testNumberOfConflictRemoteWins`

In `testNoOfConflictsStats`, you can check both `numberOfConflictLocalWins`and `numberOfConflictRemoteWins`:
* check iracManager1 for noOfConflicts==1, numberOfConflictLocalWins==1, all other counters must be zero
* check iracManager2 for noOfConflicts==1, numberOfConflictRemoteWins==1, all other counters must be zero

Reply:
Push the changes